### ISSUE TITLE
ignore.files config should also work for paths

### ISSUE THREAD

**Issue by stefanzweifel** [stefanzweifel/laravel-stats#17]
The package currently ignores file paths which are set in the [config](https://github.com/stefanzweifel/laravel-stats/blob/master/config/laravel-stats.php#L22).
The [`ClassFinder`](https://github.com/stefanzweifel/laravel-stats/blob/master/src/ClassFinder.php#L110) set's the `notName`-method and thereby ignores file paths.

I think it would be better if you can set file paths relative to the project root.

**Comment by jerguslejko** [comment]
I suppose it's better **not** using relative paths. This just adds the complexity of having to resolve those paths and does not really buy the end user anything.

**Comment by stefanzweifel** [comment]
@jerguslejko Would you suggest keeping the feature as is? After testing the package in my own apps I really can't imagine why you would ever want to ignore a specific file.

Most of the files throwing errors while running `artisan stats` are top-level files like `_ide_helper.php` or `server.php` or `deploy.php` ([deployer](http://deployer.org))

**Comment by jerguslejko** [comment]
Personally, I would write the config file as follows:

```php
return [
    'ignore' => [
        'folders' => [
            base_path('app/Http/Controllers'),
        ],
        'files' => [
            base_path('helpers.php')
        ],
    ],
];
```

This approach removes any ambiguity caused by using relative paths (Should I write out the full path? just relative? relative to what?). 

**Also**, this approach would simplify the `ClassFinder` class, as it wouldn't have to care about resolving the paths. (https://github.com/stefanzweifel/laravel-stats/blob/master/src/ClassFinder.php#L26)

**Comment by stefanzweifel** [comment]
Great idea! I always forget that you can use helpers in config files.

I think it would also be better to add non-Laravel files like `_ide_helper.php` to the default config instead of hardcoding it [here](https://github.com/stefanzweifel/laravel-stats/blob/master/src/ClassFinder.php#L156)

**Comment by jerguslejko** [comment]
Why not move all of these hardcoded values to the config file? 

If users don't publish it, they won't see a difference. If they publish it, they will have much better control over what is actually ignored. **MAYBE** someone would like to have their `tests` or `resources` folders scanned as well.

**Comment by jerguslejko** [comment]
Also, you'll get rid of `getFoldersToIgnore` and `getFilesToIgnore` methods.

**Comment by jerguslejko** [comment]
Now that I'm thinking about it, what if we added a `base_path` option to the config file?

This array basically says: _"Include just `app` folder"_.

```php
$defaultIgnoredFolders = [
    'bootstrap',
    'config',
    'public',
    'resources',
    'routes',
    'storage',
    'tests',
    'vendor',
];
```

**Comment by jerguslejko** [comment]
Also,

no need to ignore these files as they live outside the `/app` dir.

```php
$defaultFilesToIgnore = [
    '*.html',
    '*twig*',
    '*.blade.php',
    'blade.php',
    'server.php',
    '_ide_helper.php',
];
```

Question is, how likely is it for someone to have code outside the `app` dir? Adding additional directories could a feature in the next release.

**Comment by stefanzweifel** [comment]
🙋‍♂️ 

I organize my apps quite different and have the application-code in another folder than `app` but I think it's a good idea.

I will go through the code and see what we can do here.

**Comment by jerguslejko** [comment]
I have something in mind, let me see

**Comment by stefanzweifel** [comment]
Closed in #42.
